core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -833,7 +851,7 @@ object UnitActions {
         if (unit.isTransported) return false
         val tileImprovement = tile.getTileImprovement()
         // City ruins, Ancient Ruins, Barbarian Camp, City Center marked in json
-        if (tileImprovement == null || tileImprovement.hasUnique(UniqueType.Unpillagable)) return false
+        if ((tileImprovement == null || tileImprovement.hasUnique(UniqueType.Unpillagable)) && tile.roadStatus == RoadStatus.None) return false
```

Comment:
Does this actually work? I'd imagine that if you have a tile with both a road and an improvement, you can now pillage this tile as the road can be pillaged, but as improvements are pillaged first, the improvement is pillaged instead of the road

Reply:
This function is just to T/F if you could pillage on this tile

Reply:
I understand, but I believe it now returns `true` when the tile has an improvement that cannot be pillaged but it has a road, and in those cases the improvement will be pillaged instead of the road

Reply:
Gotcha. I see it now. Ok